util.ts

```diff
@@ -0,0 +1,37 @@
+import type { Request } from "express";
+
+/**
+ * Merges multiple URLSearchParams or FormData instances into the first one passed.
+ */
+export function mergeParams(
+  destination: URLSearchParams,
+  ...sources: URLSearchParams[]
+): URLSearchParams;
+export function mergeParams(
+  destination: FormData,
+  ...sources: FormData[] | URLSearchParams[]
+): FormData;
+export function mergeParams(
+  destination: URLSearchParams | FormData,
+  ...sources: URLSearchParams[] | FormData[]
+) {
+  for (const source of sources) {
+    for (const [k, v] of source.entries()) {
+      // Skip file inputs in case of FormData
+      if (typeof v === "string") destination.set(k, v);
+    }
+  }
+  return destination;
+}
+
+/**
+ * Merges an Express request's body and query params into one URLSearchParams object.
+ */
+export function mergeRequestParams(req: Request) {
+  const queryParams = new URLSearchParams(req.url.slice(req.url.indexOf("?")));
```

Comment:
if `req.url` doesn't contain a `?`, the result will not be right. i.e. `queryParams` will not be empty.